### PR TITLE
replaced crypto/sha with https://github.com/minio/sha256-simd

### PR DESCRIPTION
**Pull request by utukJ**
Signed-off-by: utukj <utukphd@gmail.com>

* [ ] Change is not relevant to the end user.

## Changes

Replaced crypto/sha package with https://github.com/minio/sha256-simd

### CODE REVIEW

File: internal/cortex/chunk/schema_util.go [hunk 1]
```diff
@@ -13,6 +12,8 @@ import (
 	"strings"
 	"sync"
 
+	"github.com/minio/sha256-simd"
+
 	"fmt"
```
Comment (yeya24) [inline internal/cortex/chunk/schema_util.go:17]: Can you group the imports?

File: pkg/block/metadata/hash.go [hunk 2]
```diff
@@ -4,13 +4,14 @@
...
 
 import (
-	"crypto/sha256"
 	"encoding/hex"
 	"fmt"
 	"io"
 	"os"
 	"path/filepath"
 
+	"github.com/minio/sha256-simd"
```
Comment (yeya24) [inline pkg/block/metadata/hash.go:13]: ditto, let's merge this with the section starting from line 15

File: pkg/reloader/reloader.go [hunk 3]
```diff
@@ -6,57 +6,56 @@
...
 // Reloader type is useful when you want to:
 //
-// 	* Watch on changes against certain file e.g (`cfgFile`).
-// 	* Optionally, specify different output file for watched `cfgFile` (`cfgOutputFile`).
-// 	This will also try decompress the `cfgFile` if needed and substitute ALL the envvars using Kubernetes substitution format: (`$(var)`)
-// 	* Watch on changes against certain directories (`watchedDirs`).
+//   - Watch on changes against certain file e.g (`cfgFile`).
+//   - Optionally, specify different output file for watched `cfgFile` (`cfgOutputFile`).
+//     This will also try decompress the `cfgFile` if needed and substitute ALL the envvars using Kubernetes substitution format: (`$(var)`)
+//   - Watch on changes against certain directories (`watchedDirs`).
```
Comment (yeya24) [inline pkg/reloader/reloader.go:12]: Umm, why these are changed? Let's update them back

File: pkg/reloader/reloader.go [hunk 4]
```diff
@@ -69,6 +68,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/minio/sha256-simd"
```
Comment (yeya24) [inline pkg/reloader/reloader.go:71]: Same grouping issue

Comment (utukJ) [reply pkg/reloader/reloader.go:71]: oops. I think a go fmt extension did that.

**Comment by bwplotka** [comment]
Flake, so restarted CI job: 
image

**Comment by bwplotka** [comment]
I know we talked about testing this with other OS-es: https://github.com/thanos-io/thanos/issues/5722#issuecomment-1263264577, however testing manually anything is not sustainable. If our CI is green, we should merge it. If image build is failing later on on main, we should add extra check for that in the PR CI - it's not sustainable for community to manually test things like that, especially as likely no one runs Thanos on freeBSD or ppc64le and we don't claim full support for that (yet).

**Review by bwplotka — APPROVED** [review]
LGTM, thanks!

**Comment by bwplotka** [comment]
For future - we should benchmark if this extra deps was even worth it 🙈

**Comment by bwplotka** [comment]
cc https://github.com/thanos-io/thanos/issues/5764

**Comment by dswarbrick** [comment]
Is a cryptographically-strong digest _really_ needed for this use case? If the goal is just to have a high quality but also _fast_ hash, maybe it makes sense to look at https://cyan4973.github.io/xxHash/.

**Pull request merged**